apps/next/app/server/clerk/index.ts

```diff
@@ -24,21 +26,26 @@ export async function mustGetClerkUser(): Promise<ClerkUser> {
   return clerkUser;
 }
 
+type ParamsForFindCreateClerkOrganization = {
+  organizationId: string | null | undefined;
+  userFirstName: string;
+  createdByClerkId: string;
+};
 export async function findCreateClerkOrganizationCreatedByUser({
-  name,
+  organizationId,
+  userFirstName,
   createdByClerkId,
-}: {
-  name: string;
-  createdByClerkId: string;
-}): Promise<ClerkOrganization> {
-  const orgs = await clerkClient.users.getOrganizationMembershipList({
-    userId: createdByClerkId,
-  });
-  for (const org of orgs) {
-    if (org.organization.name === name) {
-      return org.organization;
-    }
+}: ParamsForFindCreateClerkOrganization): Promise<ClerkOrganization> {
+  const userClerkOrg = organizationId
+    ? await clerkClient.organizations.getOrganization({
+        organizationId,
+      })
+    : undefined;
+  if (userClerkOrg) {
+    return userClerkOrg;
```

Comment:
I don't really like this new way of doing it... it's confusing to have an optional organizationId param. Why couldn't we keep the original logic and add a check for who created the clerk org?

```ts
export async function findCreateClerkOrganizationCreatedByUser({
  name,
  createdByClerkId,
}: {
  name: string;
  createdByClerkId: string;
}): Promise<ClerkOrganization> {
  const orgs = await clerkClient.users.getOrganizationMembershipList({
    userId: createdByClerkId,
  });
  for (const org of orgs) {
    if (org.organization.name === name && org.organization.createdBy === createdByClerkId) {
      return org.organization;
    }
  }
  return await clerkClient.organizations.createOrganization({
    name,
    createdBy: createdByClerkId,
  });
}
```

Reply:
Never mind. I've reverted this file to its prior state, except for some changes needed after doing the Clerk upgrade.